.github/workflows/release-branch-pull-request.yml

```diff
@@ -0,0 +1,39 @@
+name: Build project
+on:
+    pull_request:
+      branches:
+        - 'release-**'
+    workflow_dispatch:
+jobs:
+    build:
+        # The type of runner that the job will run on
+        runs-on: ubuntu-latest
+    
+        # Steps represent a sequence of tasks that will be executed as part of the job
+        steps:
+          - name: Checks-out repository under $GITHUB_WORKSPACE
+            uses: actions/checkout@v2
+            with:
+              path: go/src/github.com/${{github.repository}}
+    
+          - name: Cache Go modules
+            uses: actions/cache@v2
+            with:
+              path: ${HOME}/go/pkg/mod
+              key: ${{ runner.os }}-build-${{ hashFiles('**/go.sum') }}
+              restore-keys: |
+                ${{ runner.OS }}-build-${{ env.cache-name }}-
+                ${{ runner.OS }}-build-
+                ${{ runner.OS }}
+    
+          - name: Setup Golang
+            uses: actions/setup-go@v2
+            with:
+              # NOTE: Keep the version in sync with Go toolchain in WORKSPACE.
+              go-version: '1.19.3'
+    
+          # Setup the run environment and run CI test suite
+          - name: Go build
+            run: |
+              cd ${GITHUB_WORKSPACE}/go/src/github.com/${GITHUB_REPOSITORY}
+              GOPATH=${GITHUB_WORKSPACE}/go go build ./...
```

Comment:
@ahadas do we want to build the images as well?

Reply:
yeah, that would make sense

Reply:
but we can do it separately